### PR TITLE
Consistent APIs for Database, Fido and VSO

### LINKED ISSUE #1703: Make net modules consistent in the action verbs they use
Fido uses `search` and `fetch` and vso uses `query` and `get`. Let's be consistent.

### PR DESCRIPTION
**Pull request by nabobalis**
closes #1703.
This changes `query` to `search` where possible in code and some documentation. 
This is passing local tests but I want to make sure. 

Part two will be `get` to `fetch`.

Part three would be a changelog.

### CODE REVIEW

**Comment by Cadair** [comment]
@pep8speaks quiet

**Comment by nabobalis** [comment]
I think that is everything. I do feel like I overlooked something clearly obvious.

File: examples/goes_hek_m25.py [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
 ###############################################################################
 # Finally lets plot everything together
 
 goes.peek()
-plt.axvline(parse_time(flares_hek[0].get('event_peaktime')))
-plt.axvspan(parse_time(flares_hek[0].get('event_starttime')), parse_time(flares_hek[0].get('event_endtime')),
-            alpha=0.2, label=flares_hek[0].get('fl_goescls'))
+plt.axvline(parse_time(flares_hek[0].fetch('event_peaktime')))
+plt.axvspan(parse_time(flares_hek[0].fetch('event_starttime')), parse_time(flares_hek[0].fetch('event_endtime')),
+            alpha=0.2, label=flares_hek[0].fetch('fl_goescls'))
```
Comment (nabobalis) [inline examples/goes_hek_m25.py]: I suspect here I may have messed up.

File: sunpy/database/database.py [hunk 2]
```diff
@@ -627,6 +627,11 @@ def query(self, *query, **kwargs):
 
         return sorted(db_entries, key=operator.attrgetter(sortby))
 
+    @deprecated('0.8', alternative='database.search')
+    def query(self, *query, **kwargs):
+        __doc__ = self.search.__doc__
```
Comment (Cadair) [inline sunpy/database/database.py]: does this actually work if you run sphinx or `help()`?

File: sunpy/net/jsoc/jsoc.py [hunk 3]
```diff
@@ -100,7 +101,7 @@ class JSOCClient(object):
     >>> from sunpy.net import jsoc
     >>> from sunpy.net import attrs as a
     >>> client = jsoc.JSOCClient()
-    >>> response = client.query(a.Time('2014-01-01T00:00:00', '2014-01-01T01:00:00'),
+    >>> response = client.qsearchuery(a.Time('2014-01-01T00:00:00', '2014-01-01T01:00:00'),
```
Comment (Cadair) [inline sunpy/net/jsoc/jsoc.py]: wrf

File: sunpy/database/database.py (hunk 2 above)
Comment (nabobalis) [inline sunpy/database/database.py]: I was lazy and didn't use the sphinx doc link i used for the mapbase changes.

But yes, I have not actually checked it but it exists in the database code. I just built the docs so I'll check.

File: sunpy/net/jsoc/jsoc.py (hunk 3 above)
Comment (nabobalis) [inline sunpy/net/jsoc/jsoc.py]: Copy and PASTE WENT WILD.

File: sunpy/database/database.py (hunk 2 above)
Comment (nabobalis) [inline sunpy/database/database.py]: The answer is no.

Comment (nabobalis) [inline sunpy/database/database.py]: I would assume that __doc__ would work for help but I just removed them all so...

File: sunpy/net/vso/__init__.py [hunk 4]
```diff
@@ -3,10 +3,7 @@
 # for exposure to from sunpy.net.vso import *
 from sunpy.net.vso.vso import VSOClient, InteractiveVSOClient, QueryResponse
 
-# for, e.g.,
-# >>> from sunpy.net import vso
-# >>> vso.search(...)
-# >>> vso.get(...)
+# Delete this in 0.9
```
Comment (Cadair) [inline sunpy/net/vso/__init__.py]: after 0.9

Comment (nabobalis) [inline sunpy/net/vso/__init__.py]: after? Isn't it only one version?

File: CHANGELOG.md [hunk 5]
```diff
@@ -82,7 +82,8 @@ Latest
 * Added VSO query result block level caching in the database module. This
   prevents redownloading of files which have already been downloaded.
   Especially helpful in case of overlapping queries.
-
+* All Clients that are able to search and download data now have a uniform API that is `search` and `fetch`. The older functions are still there but are deprecated for
```
Comment (dpshelio) [inline CHANGELOG.md]: Isn't the line too long? The others are cut.

File: sunpy/spectra/sources/callisto.py [hunk 6]
```diff
@@ -93,6 +93,11 @@ def query(start, end, instruments=None, url=DEFAULT_URL):
             opn.close()
         day += _DAY
 
+@deprecated('0.8', alternative='callisto.search')
+def query(start, end, instruments=None, url=DEFAULT_URL):
+    __doc__ = search.__doc__
+    return search(start, end, instruments=instruments, url=url)
```
Comment (dpshelio) [inline sunpy/spectra/sources/callisto.py]: Probably I'll have to propagate this to sunpy/spectra, or move callisto to sources in unidown.

**Comment by nabobalis** [comment]
One last update, once CI passes I will merge.

**Comment by Cadair** [comment]
online build timedout on some tests.

**Pull request merged**